tests/integration_tests/functional/test_snapshot_basic.py

```diff
@@ -104,6 +138,47 @@ def _test_seq_snapshots(context):
         microvm.kill()
 
 
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_patch_drive_snapshot(network_config,
+                              bin_cloner_path):
+    """Test scenario: 5 full sequential snapshots."""
+    logger = logging.getLogger("snapshot_sequence")
+
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Testing matrix:
+    # - Guest kernel: Linux 4.9/4.14
+    # - Rootfs: Ubuntu 18.04
+    # - Microvm: 2vCPU with 512 MB RAM
+    # TODO: Multiple microvm sizes must be tested in the async pipeline.
+    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="2vcpu_512mb"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels(keyword="4.14"))
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="ubuntu"))
+
+    # Create a test context and add builder, logger, network.
+    test_context = TestContext()
+    test_context.custom = {
+        'builder': MicrovmBuilder(bin_cloner_path),
+        'network_config': network_config,
+        'logger': logger,
+        'snapshot_type': SnapshotType.FULL,
+        'seq_len': 1,
+        'check_patch_drive': True
+    }
+
+    # Create the test matrix.
+    test_matrix = TestMatrix(context=test_context,
+                             artifact_sets=[
+                                 microvm_artifacts,
+                                 kernel_artifacts,
+                                 disk_artifacts
+                             ])
+
+    test_matrix.run_test(_test_seq_snapshots)
```

Comment:
Looks like `_test_seq_snapshots` is quickly becoming a "god" test fn 😄 . I would just create another separate simple test. 

Or do you think it is worth to complicate it and avoid duplicating code ?

Reply:
I read it as a general (`god`) function that will run different `context`s. To me it doesn't look too big or complex right now, but we could duplicate the common parts and split it in specialized functions if you think it makes more sense.

Reply:
The part where a second scratch disk is added is not very general. I'm in favor of moving the common parts to separate methods, and having different test functions for each usecase instead of checking a lot of context flags in `_test_seq_snapshots`. For example, for verifying that patching the block device works we don't need to also run `fio`.

Reply:
done.